controllers/etcd_custodian_controller.go

```diff
@@ -168,6 +384,78 @@ func (ec *EtcdCustodian) updateEtcdStatus(ctx context.Context, logger logr.Logge
 	})
 }
 
+func (ec *EtcdCustodian) reconcileJob(ctx context.Context, logger logr.Logger, etcd *druidv1alpha1.Etcd, renderedChart *chartrenderer.RenderedChart) (*batchv1.Job, error) {
+	logger.Info("Reconcile etcd compaction job")
+
+	var job batchv1.Job
+	err := ec.Get(ctx, types.NamespacedName{Name: getJobName(etcd), Namespace: etcd.Namespace}, &job)
+
+	if err == nil {
+		// Delete job and reque if the last job failed
+		if job.Status.Failed > 0 {
```

Comment:
Are you sure that this condition is sufficient to delete the entire job? What about retries? Then there will be `Failed` executions but you probably interrupt an active compaction process.

Reply:
I think, I can mention the number of retry in job spec and I kept it as default which is 1. Let me check and confirm